packages/admin-ui-core/src/createTheme.ts

```diff
@@ -235,3 +240,86 @@ export function objectToVars(obj: Record<string, any>, parent = '') {
 
   return vars
 }
+
+/**
+ * Returns the value as a CSS Variable if it matches a value from the colors theme, otherwise it will return the arbitrary value.
+ * @example
+ * const theme = {
+ *   colors: {
+ *     black: '#000'
+ *   },
+ *   fg: {
+ *     primary: '#000',
+ *     secondary: '#dedede',
+ *   }
+ * }
+ *
+ * resolveValue('#000', 'fg', theme)
+ * // returns: var(--admin-ui-colors-black)
+ *
+ * resolveValue('#dedede', 'fg', theme)
+ * // returns: #dedede
+ */
+function resolveValue(value: any, ruleId: string, theme: Record<string, any>) {
+  if (!constants.colorNamespaces.includes(ruleId)) return value
+
+  const colors = get(theme, 'colors', {})
+
+  const colorsKeys = Object.keys(colors)
+
+  let result = value
+
+  colorsKeys.forEach((key) => {
+    const colorValue = get(colors, key)
+
+    if (result.includes(colorValue)) {
+      result = result.replaceAll(colorValue, toVarValue(`colors-${key}`))
+    }
+  })
+
+  return result
+}
+
+/**
+ * Parses an theme recursivelly to css variables, joining the paths
+ * @example
+ * convertToVars({
+ *  colors: {
+ *    blue: 'blue',
+ *    yellow: 'yellow'
+ *  },
+ *  bg: {
+ *    primary: colors.blue40,
+ *    secondary: colors.yellow40
+ *  }
+ * })
+ *
+ * // returns:
+ * {
+ *   '--admin-ui-colors-blue': 'blue',
+ *   '--admin-ui-colors-yellow': 'yellow',
+ *   '--admin-ui-bg-primary': 'var(--admin-ui-colors-blue)',
+ *   '--admin-ui-bg-secondary': 'var(--admin-ui-colors-yellow)',
+ * }
+ */
+export function generateVars<T>(obj: T, theme = {}, ruleId = '', parent = '') {
```

Comment:
**Suggestion**: Maybe changing the `parent` param name could be good, I was expecting this param to be a object, but it is the key name which represents the "parent object" or something like it.